### PR TITLE
POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by tatsuya98**
added tests for post request
added logic to errorHandling
return everything after updating instead of just the body

### CODE REVIEW

**Review by MRombi — COMMENTED** [review]
Good job overall :)

File: __tests__/api.test.js [hunk 1]
```diff
@@ -173,7 +173,12 @@ describe("/api/articles/:article_id/comments", () => {
         .expect(201)
         .then(({ body: { userComment } }) => {
           expect(userComment).toEqual({
+            comment_id: expect.any(Number),
             body: expect.any(String),
```
Comment (nc-niamh-smith) [inline __tests__/api.test.js:177]: You know what the body is! :)

File: __tests__/api.test.js [hunk 2]
```diff
@@ -190,6 +195,31 @@ describe("/api/articles/:article_id/comments", () => {
...
+        body: "Hello",
+      };
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send(testComment)
+        .expect(400)
+        .then(({ body: { message } }) => {
+          expect(message).toBe("request is missing 1 or more fields");
+        });
+    });
```
Comment (nc-niamh-smith) [inline __tests__/api.test.js:209]: nice!

File: __tests__/api.test.js [hunk 3]
```diff
@@ -190,6 +195,31 @@ describe("/api/articles/:article_id/comments", () => {
...
+        body: "hi",
+      };
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send(testComment)
+        .expect(404)
+        .then(({ body: { message } }) => {
+          expect(message).toBe("username does not exist");
+        });
+    });
```
Comment (nc-niamh-smith) [inline __tests__/api.test.js:222]: nice!

File: errorHandling/errorHandling.js [hunk 4]
```diff
@@ -9,12 +9,21 @@ exports.SqlErrorHandling = (error, request, response, next) => {
   if (error.code === "22P02") {
     response.status(400).send({ message: "Bad request" });
   }
-  if (error.code === "23503") {
+  if (error.code === "23503" && error.detail.includes("articles")) {
+    response.status(422).send({
+      message: "unable to post comment to an article that does not exist",
```
Comment (nc-niamh-smith) [inline errorHandling/errorHandling.js:14]: interesting use of the properties on the error - I really like this!

**Review by nc-niamh-smith — COMMENTED** [review]
You know what the body is when you've created the comment - make sure you can specify that nothing happens to it :)

**Pull request merged**